internal/services/containerapps/container_app_resource.go

```diff
@@ -0,0 +1,435 @@
+		},
+	}
+}
+
+func (r ContainerAppResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ContainerApps.ContainerAppClient
+
+			id, err := containerapps.ParseContainerAppID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state ContainerAppModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			model := existing.Model
+
+			if model.Properties == nil {
+				return fmt.Errorf("retreiving properties for %s for update: %+v", *id, err)
+			}
+
+			if model.Properties.Configuration == nil {
+				model.Properties.Configuration = &containerapps.Configuration{}
+			}
+
+			// Delta-updates need the secrets back from the list API, or we'll end up removing them or erroring out.
+			secretsResp, err := client.ListSecrets(ctx, *id)
+			if err != nil || secretsResp.Model == nil {
+				if secretsResp.HttpResponse == nil || secretsResp.HttpResponse.StatusCode != http.StatusNoContent {
+					return fmt.Errorf("retrieving secrets for update for %s: %+v", *id, err)
+				}
+			}
+			model.Properties.Configuration.Secrets = helpers.UnpackContainerSecretsCollection(secretsResp.Model)
+
+			if metadata.ResourceData.HasChange("revision_mode") {
+				revisionMode := containerapps.ActiveRevisionsMode(state.RevisionMode)
+				model.Properties.Configuration.ActiveRevisionsMode = &revisionMode
+			}
+
+			if metadata.ResourceData.HasChange("ingress") {
+				model.Properties.Configuration.Ingress = helpers.ExpandContainerAppIngress(state.Ingress, id.ContainerAppName)
+			}
+
+			if metadata.ResourceData.HasChange("registry") {
+				model.Properties.Configuration.Registries = helpers.ExpandContainerAppRegistries(state.Registries)
+
+			}
+
+			if metadata.ResourceData.HasChange("dapr") {
+				model.Properties.Configuration.Dapr = helpers.ExpandContainerAppDapr(state.Dapr)
+
+			}
+
+			if metadata.ResourceData.HasChange("template") {
+				if model.Properties.Template == nil {
+					model.Properties.Template = &containerapps.Template{}
+				}
+				allProbesRemoved := helpers.ContainerAppProbesRemoved(metadata)
+				if allProbesRemoved {
+					containers := *model.Properties.Template.Containers
+					containers[0].Probes = pointer.To(make([]containerapps.ContainerAppProbe, 0))
+					model.Properties.Template.Containers = &containers
+				}
+			}
+
+			if metadata.ResourceData.HasChange("secret") {
+				model.Properties.Configuration.Secrets = helpers.ExpandContainerSecrets(state.Secrets)
+			}
+
+			if metadata.ResourceData.HasChange("identity") {
+				ident, err := identity.ExpandSystemAndUserAssignedMapFromModel(state.Identity)
+				if err != nil {
+					return err
+				}
+				model.Identity = pointer.To(identity.LegacySystemAndUserAssignedMap(*ident))
```

Comment:
(same here) we're missing expand/flatten funcs for the legacy identity types?

Reply:
As discussed offline, this will be followed up with the correct application of the functions when they are released in `go-azure-helpers` (PR to follow shortly).